ibis/backends/sql/compilers/base.py

```diff
@@ -1527,6 +1532,269 @@ def visit_DropColumns(self, op, *, parent, columns_to_drop):
         )
         return sg.select(*columns_to_keep).from_(parent)
 
+    def visit_GenericInfo(self, op, *, parent, **_):
```

Comment:
Implementing these with rewrite rules just to branch on which implementation to call feels weird to me. `Describe`, `GenericDescribe`, and `FastDescribe` all contain the same attributes, the classes only exist to dispatch to a different implementation.

Why not implement these as just `visit_Describe` and use a compiler-level flag to dispatch. IIUC the deciding factor here is whether the backend implements `Unnest`? If so, you could have `visit_Describe` branch on whether the backend defines `visit_Unnest` without needing to introduce a new compiler-level switch.

Reply:
Lowering operations to be more specific, even if the attributes are the same, has some value in that it keeps complex branching out of the codebase. The classes are explicitly written in this way precisely to avoid a "checking" implementation.

Are you concerned about rewrite rule overhead? The checking approach feels like an unnecessary hack, but maybe you're of the opinion that my approach is adding new classes unnecessarily :)

Reply:
> but maybe you're of the opinion that my approach is adding new classes unnecessarily :)

Yeah, it's this. Doing it this way feels _more complex_ to me than using a simple branch in the compiler (whether that's on a new flag or on a `has_operation(Unnest)` or something else :shrug:). The rewrite rules are doing the same thing, but making it more opaque when and how it happens. I'd personally find a single implementation of `visit_Describe` that did the branching easier to read and understand than having 3 classes, 2 rewrite rules, and 2 implementations.

Reply:
Fair enough, I'm happy to look into the other approach, but there'll at least be two private methods, one for each implementation. I won't be stuffing all that junk into a single megamethod.

Reply:
Indeed, @jcrist your approach is simpler, thanks for pushing back!